### DISCUSSION TITLE
No label in kube_pod_info

### DISCUSSION THREAD

**Discussion by PierreRust** [prometheus-operator/kube-prometheus#1379] [Q&A]
Hi all,

I've installed a basic kube-prometheus exporter, with no specific configuration and I need to join some metrics with the pod's labels. 
This is usually done by joining with the `kube_pod_info` metric (from kube-state-metrics) but this metric, while present, does not contain any label. I've also check kube-state-metrics's exporter directly and it does not exposes the labels.
For example with a test pod and a app label: 
```
> kubectl  describe pod mysite-nginx-6958b75c8c-p97xv
Name:         mysite-nginx-6958b75c8c-p97xv
Namespace:    default
Priority:     0
Node:         bm-debian-1/172.16.21.11
Start Time:   Mon, 23 Aug 2021 18:51:53 +0200
Labels:       app=mysite-nginx
              pod-template-hash=6958b75c8c
...
```
The exposed metrics does not contain the app label : 
```
kube_pod_labels{namespace="default",pod="mysite-nginx-6958b75c8c-p97xv",uid="a54274d5-c433-49f9-bc18-df9b05125b95"} 1
```

I'm wondering if that may come from the way kube-state-metrics is configured by the operator, From what I see in kubernetes/kube-state-metrics#1501 `--metric-labels-allowlist=pods=[*]` should be passed to kube-state-metrics to collect all labels.

I that something that's missing the operator configuration for kube-state-metrics or am I missing something else  ?

If you think that's a bug, I can of course create an issue for this.

Thanks !

**Comment by paulfantom** [comment]
> I that something that's missing the operator configuration for kube-state-metrics or am I missing something else ?

This is by design to limit label cardinality. Users need to specify which labels they are interested in with `metric-labels-allowlist` and pass it to kube-state-metrics configuration. Example: https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/jsonnet/main.jsonnet#L272-L283
